packages/test/test-end-to-end-tests/src/test/summarizerWithLocalChanges.spec.ts

```diff
@@ -0,0 +1,181 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { DataObject, DataObjectFactory } from "@fluidframework/aqueduct";
+import { IContainer } from "@fluidframework/container-definitions";
+import { IContainerRuntimeOptions, ISummarizer } from "@fluidframework/container-runtime";
+import { IRequest } from "@fluidframework/core-interfaces";
+import { SharedMatrix } from "@fluidframework/matrix";
+import { SharedMap } from "@fluidframework/map";
+import {
+	ITestObjectProvider,
+	waitForContainerConnection,
+	summarizeNow,
+	createSummarizerFromFactory,
+} from "@fluidframework/test-utils";
+import {
+	describeNoCompat,
+	getContainerRuntimeApi,
+	itExpects,
+} from "@fluid-internal/test-version-utils";
+import { IContainerRuntimeBase, IFluidDataStoreFactory } from "@fluidframework/runtime-definitions";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { FluidDataStoreRuntime, mixinSummaryHandler } from "@fluidframework/datastore";
+import { pkgVersion } from "../packageVersion";
+
+const runtimeOptions: IContainerRuntimeOptions = {
+	summaryOptions: {
+		summaryConfigOverrides: { state: "disabled" },
+	},
+	gcOptions: { gcAllowed: true },
+};
+export const rootDataObjectType = "@fluid-example/rootDataObject";
+export const TestDataObjectType1 = "@fluid-example/test-dataStore1";
+export const TestDataObjectType2 = "@fluid-example/test-dataStore2";
+class TestDataObject2 extends DataObject {
+	public get _root() {
+		return this.root;
+	}
+	public get _context() {
+		return this.context;
+	}
+	// If this datastore created DDSes it would be even more nodes created without running gc
+}
+
+class TestDataObject1 extends DataObject {
+	public get _root() {
+		return this.root;
+	}
+
+	public get _context() {
+		return this.context;
+	}
+
+	private readonly datastoreKey = "TestDataObject2";
+
+	protected async hasInitialized() {
+		// This can be fired synchronously as well, which would cause an even worse half state
+		await this.init();
+	}
+
+	private async init() {
+		const dataObject2 = await rootDataObjectFactory.createInstance(
+			this.context.containerRuntime,
+		);
+		this.root.set(this.datastoreKey, dataObject2.handle);
```

Comment:
This is not the same scenario as what is seen in tasks component right? This will end up always failing, no matter what we do - For example, even if did a second summary attempt, this will fail. Whereas in case of tasks component, this would succeed.
Maybe adding a check that the data store does not exist, before creating a new one will make it closer to the real one?
Also, wasn't there something about this being not awaited?

Reply:
Ok, I can add a modification where this does succeed eventually.